luma/core/sprite_system.py

```diff
@@ -0,0 +1,180 @@
+# -*- coding: utf-8 -*-
+# Copyright (c) 2014-17 Richard Hull and contributors
+# See LICENSE.rst for details.
+
+import time
+from PIL import Image
+
+
+class dict_wrapper(object):
+    """
+    Helper class to turn dictionaries into objects.
+    """
+    def __init__(self, d):
+        for a, b in d.items():
+            if isinstance(b, (list, tuple)):
+                setattr(self, a, [dict_wrapper(x) if isinstance(x, dict) else x for x in b])
+            else:
+                setattr(self, a, dict_wrapper(b) if isinstance(b, dict) else b)
+
+
+class spritesheet(object):
+    """
+    A sprite sheet is a series of images (usually animation frames) combined
+    into a larger image. A dictionary is usually spread into the object
+    constructor with the following attributes:
+
+        * ``image``: a string representing the path to a sprite map image.
```

Comment:
i'd prefer to use the Sphinx/rst way of documenting parameters and attributes, what about you?